pyartcd/pyartcd/pipelines/promote.py

```diff
@@ -483,6 +484,22 @@ async def publish_client(self, working_dir, from_release_tag, release_name, arch
         CLIENT_MIRROR_DIR = f"{BASE_TO_MIRROR_DIR}/{arch}/clients/{client_type}/{release_name}"
         os.makedirs(CLIENT_MIRROR_DIR)
 
+        # Get cli pull-spec from the release
+        image_stat, cli_pull_spec = get_release_image_pullspec(f"{quay_url}:{from_release_tag}", "cli")
+        if image_stat == 0:  # image exists
+            image_info = get_release_image_info(cli_pull_spec)
+            # Retrieve the commit from image info
+            commit = image_info["config"]["config"]["Labels"]["io.openshift.build.commit.id"]
+            # URL to download the tarball from oc repo for a specific commit
+            url = f"https://github.com/openshift/oc/archive/{commit}.tar.gz"
+
+            # download the tarball
+            target_path = f"{CLIENT_MIRROR_DIR}/oc-source.tar.gz"
+            response = requests.get(url, stream=True)
+            if response.status_code == 200:
+                with open(target_path, "wb") as f:
+                    f.write(response.raw.read())
```

Comment:
you may also need to get and write sha256sum to sha256sum.txt file, it will not automatically be handled in the following code